src/Database/LSMTree/Internal/Run/Index/Compact.hs

```diff
@@ -600,6 +611,155 @@ fromChunks cs FinalChunk{..} = CompactIndex {
     , ciLargerThanPage       = fcLargerThanPage
     }
 
+{-------------------------------------------------------------------------------
+  Serialisation
+-------------------------------------------------------------------------------}
+
+-- | 32 bit aligned.
+chunkBuilder :: Chunk -> BB.Builder
+chunkBuilder Chunk {..} = VU.foldMap BB.word32LE cPrimary
```

Comment:
We could make this quite a lot faster. The most agressive approach would be to make sure the byte arrays backing the chunks are pinned, and then we could convert to `ByteString` directly for I/O (or on to builder without copying).

Or less aggressive: use `byteArrayFromTo` to give a builder that copies from the chunk vector in bulk rather than an element-by-element loop (like `foldMap` here).

Reply:
I'll give that a try.

Reply:
I implemented the `byteArrayFromTo` version and documented the possibility of a more efficient version.